added features, ways to reach to a location, implimented using google…

HemabhKamboj opened the pull request:
# Description
Implemented using following APIs google maps
-Place search API
-Place autocomplete API
-Place Detail API
-Distance Matrix API
Fixes #(68)

GouravSardana commented:
@shikharvaish28 @prabhakar267 I reviewed it. You can also review it.

shikharvaish28 commented on nomad/settings.py:
```diff
@@ -34,7 +34,7 @@
 DEBUG = (os.environ.get('RUNNING_PROD') is None)
 
-ALLOWED_HOSTS = ['localhost', '.herokuapp.com']
+ALLOWED_HOSTS = ['localhost', '127.0.0.1', '.herokuapp.com']
```
IMO this line is not needed in the PR since it is for your local setup.

shikharvaish28 commented on api/modules/ways/views.py:
```diff
@@ -0,0 +1,282 @@
+        return Response(str(e), status=status.HTTP_503_SERVICE_UNAVAILABLE)
+
+    return Response(response_string)
+
```
Remove these extra lines at the end. Just one line at the end is good enough.

shikharvaish28 commented on api/modules/ways/views.py:
```diff
@@ -0,0 +1,282 @@
+@api_view(['GET'])
+def get_ways_train(request, origin, destination):
+    """
+
```
These blank lines are not needed.

shikharvaish28 commented on api/modules/ways/views.py:
```diff
@@ -0,0 +1,282 @@
+        if place_autocomplete_api_response.status_code == 403:
+            error_message = 'Places Autocomplete API error - Permission denied/Forbidden'
+
+            # create GitHub issue to get developers' attention
```
a minute type. It should be developer's

shikharvaish28 commented on api/modules/ways/views.py:
```diff
@@ -0,0 +1,282 @@
+            error_message = 'Places Search API error - Permission denied/Forbidden'
+
+            # create GitHub issue to get developers' attention
+            make_github_issue(error_message)
```
In ```make_github_issue(error_message)``` you are passing the error message so instead of writing this very statement, again and again, make a flag which should see if any error has occurred. In case it has then passed the appropriate ```error_message``` and end it once for all.

HemabhKamboj replied:
`make_github_issue(error_message)` is only called when api response status code is 403.
So do you want that whenever google maps api response is 403,  Flag is turned to true, and whenever the flag is turned true, 'make_github_issue(error_message)' be called? And hence 'make_github_issue(error_message)' will be called only once in whole route( one function) ?

shikharvaish28 replied:
Yes. But the ```error_message``` is changing at many places so you can call ```make_github_issue(error_message)``` just once. You have called ```make_github_issue(error_message)``` several times which can be reduced to just one time.

shikharvaish28 commented on api/modules/ways/views.py:
```diff
@@ -0,0 +1,282 @@
+
+            # create GitHub issue to get developers' attention
+            make_github_issue(error_message)
+
```
No need for blank line.

shikharvaish28 commented on api/modules/ways/views.py:
```diff
@@ -0,0 +1,282 @@
+            return Response(error_message, status=status.HTTP_400_BAD_REQUEST)
+        if distance_api_response.status_code == 403:
+            error_message = 'Distance Matrix API error  - Permission denied/Forbidden'
+
```
Please remove these blank lines.

shikharvaish28 commented on api/modules/ways/views.py:
```diff
@@ -0,0 +1,282 @@
+@api_view(['GET'])
+def get_ways_flight(request, origin, destination):
+    """
+
```
Here too.

shikharvaish28 commented on api/modules/ways/views.py:
```diff
@@ -0,0 +1,282 @@
+            error_message = 'Places API error - Permission denied/Forbidden'
+
+            # create GitHub issue to get developers' attention
+            make_github_issue(error_message)
```
The same thing can be implemented using if-else as mentioned above.

shikharvaish28 reviewed (changes requested):
Thanks for working on this @HemabhKamboj 

Great work on this PR though I've mentioned some changes which are needed to be accommodated. For coding style, you can refer to the PEP8 pattern since there are a lot of blank lines which are necessary to remove.

Also, you need to change the commit message since they are too long and are not giving a clear idea of what you're trying to do. You can keep the commit message to be short such that it provides a coherent view. To explain the commit, you can use commit description.

Please change the PR according to the review so that I can run you PR locally on my machine to do further test.

HemabhKamboj commented:
ok, will correct all these

HemabhKamboj commented:
PEP8 states 79 characters per line, but Travis Build Flake 8 passed even when there more characters. And in my code, there are many lines having nore than 79 characters. Should I fix them too?

HemabhKamboj committed:
added feature, user can now search for ways to reach to particular location (#157)
modified: api/urls.py. Added required routes for the API call.
added file: ways/views.py. Defined 3 ways to reach a particular location, train, flight and road, taking source, destination and/or mode as arguements, implemented using Google Maps APIs.

shikharvaish28 commented on requirements.txt:
```diff
@@ -9,7 +9,7 @@ gunicorn==19.8.1
 mccabe==0.6.1
 oauthlib==2.1.0
-psycopg2==2.7.4
+psycopg2-binary==2.7.7
```
IMO this can be made a separate issue for beginners since your commit is in no way related to this change.

shikharvaish28 replied:
Remove this line from your commit.

shikharvaish28 reviewed (changes requested):
Great work @HemabhKamboj 👍 

The code LGTM. The last thing you need to do is to ```amend``` the commit message and description to be short and in a proper format which can give a coherent idea of what exactly does the commit consists of. I would approve the PR once this is done 😄

shikharvaish28 reviewed (approved):
I am approving it as of now but please remove the line I have mentioned below. I am creating another issue for that and assigning it to you.

shikharvaish28 commented:
@HemabhKamboj  https://github.com/project-travel-mate/server/issues/158

shikharvaish28 commented:
@Swati4star @prabhakar267 IMO this PR is GTG.

prabhakar267 commented:
@HemabhKamboj great work on PR. But I will have to close it without merging. Please see my comment on issue https://github.com/project-travel-mate/server/issues/68#issuecomment-485113053.
